Spryker/Sniffs/Testing/MockSniff.php

```diff
@@ -0,0 +1,257 @@
+<?php
+
+/**
+ * MIT License
+ * For full license information, please view the LICENSE file that was distributed with this source code.
+ */
+
+namespace Spryker\Sniffs\Testing;
+
+use PHP_CodeSniffer\Files\File;
+use SlevomatCodingStandard\Helpers\FunctionHelper;
+use SlevomatCodingStandard\Helpers\ReturnTypeHint;
+use Spryker\Sniffs\AbstractSniffs\AbstractSprykerSniff;
+
+/**
+ * Ensures no wrong usage of MockObject return typehint or docblock return annotation.
+ *
+ * @author Mark Scherer
+ * @license MIT
+ */
+class MockSniff extends AbstractSprykerSniff
+{
+    protected const MOCK_OBJECT = '\PHPUnit\Framework\MockObject\MockObject';
+
+    /**
+     * @inheritDoc
+     */
+    public function register(): array
+    {
+        return [T_FUNCTION];
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function process(File $phpcsFile, $stackPtr)
+    {
+        if (!$this->isTest($phpcsFile, $stackPtr)) {
+            return;
+        }
+
+        $returnTypeHint = FunctionHelper::findReturnTypeHint($phpcsFile, $stackPtr);
+        $docBlockReturnTypes = $this->getDocBlockReturnTypes($phpcsFile, $stackPtr);
+
+        $this->assertDocBlockReturnAnnotation($phpcsFile, $stackPtr, $docBlockReturnTypes, $returnTypeHint);
+        $this->assertNoReturnTypehint($phpcsFile, $stackPtr, $returnTypeHint, $docBlockReturnTypes);
+    }
+
+    /**
+     * @param \PHP_CodeSniffer\Files\File $phpcsFile
+     * @param int $stackPtr
+     * @param \SlevomatCodingStandard\Helpers\ReturnTypeHint|null $returnTypeHint
+     * @param string[] $docBlockReturnTypes
+     *
+     * @return void
+     */
+    protected function assertNoReturnTypehint(File $phpcsFile, int $stackPtr, ?ReturnTypeHint $returnTypeHint, array $docBlockReturnTypes): void
+    {
+        if (!$returnTypeHint || $returnTypeHint->getTypeHint() !== 'MockObject') {
+            return;
+        }
+
+        $fix = $phpcsFile->addFixableError('MockObject must not be typehinted.', $returnTypeHint->getStartPointer(), 'InvalidTypehint');
+        if (!$fix) {
+            return;
+        }
+
+        $this->removeReturnTypeHint($phpcsFile, $stackPtr, $returnTypeHint);
+    }
+
+    /**
+     * @param \PHP_CodeSniffer\Files\File $phpcsFile
+     * @param int $stackPtr
+     * @param \SlevomatCodingStandard\Helpers\ReturnTypeHint $returnTypeHint
+     *
+     * @return void
+     */
+    protected function removeReturnTypeHint(File $phpcsFile, int $stackPtr, ReturnTypeHint $returnTypeHint): void
+    {
+        $colonPointer = $phpcsFile->findPrevious(T_COLON, $returnTypeHint->getStartPointer(), $stackPtr);
+        if (!$colonPointer) {
+            return;
+        }
+
+        $phpcsFile->fixer->beginChangeset();
+        $endPtr = $returnTypeHint->getEndPointer();
+
+        for ($i = $colonPointer; $i <= $endPtr; $i++) {
```

Comment:
Consider renaming `colonPointer` above to `pointerToPreviousColon`

```suggestion
        for ($colonPointer = $pointerToPreviousColon; 
```